### PR TITLE
Add Archives APIs

### PR DESCRIPTION
**Pull request by FledgeXu**
# Goal
The goal of this PR is to add a series of APIs that manipulate Archives
# Abstract
1. Added API for creating an Archive
2. Added an API to get information about an Archive.
3. added an API to modify the metadata of an Archive.

**Notice: This PR does not include any interaction with ZimFarm.**

### CODE REVIEW

File: backend/api/routes/archives.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+@router.patch(
+    "/{project_id}/archives/{archive_id}",
+    status_code=HTTPStatus.NO_CONTENT,
+)
+async def update_archive(
+    archive_request: ArchiveRequest,
+    archive: Archive = Depends(validated_archive),
+    project: Project = Depends(validated_project),
+    session: Session = Depends(gen_session),
+):
```
Comment (FledgeXu) [inline backend/api/routes/archives.py:130]: For now, I haven't add a support for Illustration image. I think I should, But I don't know what's the best way to deal with this part.

Comment (rgaudin) [reply backend/api/routes/archives.py:130]: Yes ZIM Illustration is important. You may use a dedicated endpoint if you think that makes sense UX-wise.

It's a regular file upload for which you:
- check that its declared as a std image
- check uploaded filesize (fail if too big)
- convert to PNG using scraperlib
- resize to 48x48 using scraperlib
- store as base64. That's appropriate for our small 48px PNG. We could store as bytea but we'd have to serialize for sending and we don't have much use of this in the server.

In the UI, the process will be mirrored:
- check with JS the filetype and filesize (constants to config the max illustration size –– 2MiB?)
- upload ; alert if it failed, display if OK

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thank you ; 

This current design cannot support the UX we designed though. It's not clear what trigger a POST on archives and whether/when it is being requested.

We want the user to be able to set all this information at any time and then, once ready, request the _materialization_ of the ZIM.

This implies a couple thing:
- An archive should be immediately created when a Project is created.
- Archive can be updated (its config)
- When _Requesting_ a ZIM, a couple things happen:
  - the collections.json file is generated and uploaded (and its URL set on the DB)
  - the status of the Archive changes
  - the task is requested on ZF (and taskId should be stored on DB)
  - A new, Archive should be created, with a similar config.
- Via ZF callback, Archive is updated: status and maybe download_url and filesize

This is because we chose to store the archive related metadata on the Archive table only. We could also duplicate it all on the project and only create Archive upon request (copying values)

This means that Archives would have statuses as:
- PENDING (it's in DB but not requested ; can be modified)
- REQUESTED (it has been ZF-requested ; cant be modified by user, awaiting callback from ZF)
- READY (ZF succeeded, it now has a download_url and filesize)
- FAILED (ZF task failed, cant be downloaded)

File: backend/api/routes/archives.py (hunk 1 above)
Comment (FledgeXu) [reply backend/api/routes/archives.py:130]: Sorry for late reply, new semester really catch me. 😢
I will create a separate endpoint for this.

File: backend/tests/conftest.py [hunk 2]
```diff
@@ -110,6 +109,22 @@ def test_file():
...
+    return (
+        b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00"
+        b"\x01\x08\x04\x00\x00\x00\xb5\x1c\x0c\x02\x00\x00\x00\x0bIDATx\xdacd`\x00\x00\x00\x06\x00\x020\x81\xd0/\x00\x00\x00\x00IEND\xaeB`\x82"
+    )
+
+
+@pytest.fixture
+def test_jpeg_image():
+    return (
+        b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00"
```
Comment (rgaudin) [inline backend/tests/conftest.py:123]: Doesn't look like a JPEG image (it has the PNG magic number)

Comment (rgaudin) [reply backend/tests/conftest.py:123]: Actually I think it's not used

**Comment by FledgeXu** [comment]
Can not pass the pyright, the I think it's a bug of zimscraperlib.
See: https://github.com/openzim/python-scraperlib/issues/112

**Review by rgaudin — APPROVED** [review]
OK thank you ; let's merge this

File: backend/api/routes/archives.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+async def create_archive(
+    archive_request: ArchiveRequest,
+    project: Project = Depends(validated_project),
+    session: Session = Depends(gen_session),
+) -> ArchiveModel:
+    """Create a pre-request archive"""
+    now = datetime.datetime.now(tz=datetime.UTC)
+    new_archive = Archive(
+        filename=archive_request.filename
+        if archive_request.filename
```
[the remark this answers is not in the document]
Comment (rgaudin) [reply backend/api/routes/archives.py]: added

**Pull request merged**